lms/djangoapps/verify_student/tests/test_views.py

```diff
@@ -885,12 +895,14 @@ def test_create_order_already_verified(self):
 
         # Verify that the information will be sent to the correct callback URL
         # (configured by test settings)
-        data = json.loads(response.content)
-        self.assertEqual(data['override_custom_receipt_page'], "http://testserver/shoppingcart/postpay_callback/")
-
-        # Verify that the course ID and transaction type are included in "merchant-defined data"
-        self.assertEqual(data['merchant_defined_data1'], unicode(self.course.id))
-        self.assertEqual(data['merchant_defined_data2'], "verified")
+        self._check_payment_data(
+            response,
+            {
+                'override_custom_receipt_page': "http://testserver/shoppingcart/postpay_callback/",
+                'merchant_defined_data1': unicode(self.course.id),
+                'merchant_defined_data2': "verified",
```

Comment:
I agree with paring down the test such that it verifies the expected behavior.

That said, looking at this test and others like it more closely, I don't think there should be a difference in behavior at this point if the user is verified. Formerly, users couldn't make it to payment unless they were already verified. Currently, this part of the flow should behave the same for verified and non-verified users alike.